mails/themes/classic/core/order_conf.html.twig

```diff
@@ -48,19 +48,32 @@
           <th bgcolor="#f8f8f8" style="border:1px solid #D6D4D4;background-color: #fbfbfb;color: #333;font-family: Arial;font-size: 13px;padding: 10px;">{{ 'Quantity'|trans({}, 'Emails.Body', locale)|raw }}</th>
           <th bgcolor="#f8f8f8" style="border:1px solid #D6D4D4;background-color: #fbfbfb;color: #333;font-family: Arial;font-size: 13px;padding: 10px;" width="17%">{{ 'Total price'|trans({}, 'Emails.Body', locale)|raw }}</th>
         </tr>
-        {% if templateType == 'html' %}
-{products}
-{% endif %}
-        {% if templateType == 'txt' %}
-{products_txt}
-{% endif %}
-
-        {% if templateType == 'html' %}
-{discounts}
-{% endif %}
-        {% if templateType == 'txt' %}
-{discounts_txt}
-{% endif %}
+        <!-- delete -->
```

Comment:
So if I understand correctly the purpose of the PR, you need this template to be html valid because the missing tr/td messed with the template saving process in TinyMCE

So the idea is to add this tr/td to make TinyMCE (or its related controller) happy, except this extra tr/td shouldn't be present because the variables are already replaced with appropriate values So the SwiftCommentDeletePlugin is responsible for removing these extra parts when the Mail is sent?

Is that it? 😅

Reply:
Well, this is the idea... We couldn't find another solution that this ugly one :/

Reply:
Yeah I see the idea, I understand that you had no other choice But it still seems a bit unstable I'm looking for why this was changed Because in the original template the tr/td were present https://github.com/PrestaShop/PrestaShop/blob/69f840d7f626769431c9c0ae9ad8ef1a327571c0/mails/en/order_conf.html#L116

So I'm searching why they were moved into the template files https://github.com/PrestaShop/PrestaShop/blob/802b8b0a1dd58786c9ef00586b9d95a4574700bd/mails/_partials/order_conf_product_list.tpl#L26

Because I think ideally the tr/td should always be in the template file so that it remains valid

Reply:
So I don't see any structural difference between the original products template
https://github.com/PrestaShop/PrestaShop/blob/1.7.6.x/mails/en/order_conf_product_list.tpl

And the new one:
https://github.com/PrestaShop/PrestaShop/blob/1.7.7.x/mails/_partials/order_conf_product_list.tpl

However I think I understand why I changed it, the problem was that the template included a new `<tr>` in what was a `<td>` not respecting the creation of a new `table` I believe it created some display issues because although the static template with the variable `{products}` was HTML compliant, the generated one was not any more

Reply:
Ok I found the PR where this was introduced https://github.com/PrestaShop/PrestaShop/pull/13481
To fix this issue https://github.com/PrestaShop/PrestaShop/issues/13433

As you can see in the screenshots the list of product was not correctly displayed because the table/tr/td structure was not respected (so basiacally first closing tr in the list closed the inital tr and it messed up everything)

Reply:
I have a proposition that could work I think without relying on "ghost code" that is supposed to be removed later. We could init a new table inside the products cell, this way the order_conf_product_list.tpl can integrate its `tr` list smoothly

```html
<tr>
          <td style="border:1px solid #D6D4D4;padding:7px 0;text-align:center;" colspan="5">
              <table>
                  {products}
              </table>
          </td>
</tr>
```

But then I guess this would be considered invalid HTML again by the controller?

Reply:
You got it right @jolelievre! I pretty sure I already tried your suggestion but then, columns were not aligned with the table header `th`...